src/components/forms/ProxyDeviceInput.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useState } from "react";
+import { Input, Select } from "@canonical/react-components";
+
+interface Props {
+  onChange?: (type?: string, address?: string, port?: string) => void;
+  value?: string;
+  isConnect?: boolean;
+  isNatEnabled?: string;
+}
+
+const ProxyDeviceInput: FC<Props> = ({
+  onChange,
+  value,
+  isConnect,
+  isNatEnabled,
+}) => {
+  const [inheritedType, inheritedAddress, inheritedPort] =
+    value?.split(":") || [];
+
+  const [type, setType] = useState(inheritedType || "");
+  const [address, setAddress] = useState(inheritedAddress || "");
+  const [port, setPort] = useState(inheritedPort || "");
+
+  const AddressTypeOptions = [
```

Comment:
```suggestion
  const proxyAddressTypeOptions = [
```
I think this can move into the instance options file.